mypyc/lib-rt/getargs.c

```diff
@@ -1197,8 +1208,15 @@ vgetargskeywords(PyObject *args, PyObject *kwargs, const char *format,
         }
     }
 
-    if (len > STATIC_FREELIST_ENTRIES) {
-        freelist.entries = PyMem_NEW(freelistentry_t, len);
+    if (*format == '%') {
+        p_args = va_arg(*p_va, PyObject **);
+        p_kwargs = va_arg(*p_va, PyObject **);
+        format++;
+    }
+
+    max_freelist_entries = len + !!p_args + !!p_kwargs;
```

Comment:
Aw :P

I decided to rework the memory freeing in the error cases with a goto/label instead of using the freelist since it is cleaner, a little faster, and better fits a future I envision where I rip out all of the freelist code from our copy of getargs